packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -0,0 +1,1164 @@
+///     },
+///     selectedIndex: _currentPageIndex,
+///     destinations: [
+///       NavigationDestination(
+///         icon: Icon(Icons.explore),
+///         label: 'Explore',
+///       ),
+///       NavigationDestination(
+///         icon: Icon(Icons.commute),
+///         label: 'Commute',
+///       ),
+///       NavigationDestination(
+///         selectedIcon: Icon(Icons.bookmark),
+///         icon: Icon(Icons.bookmark_border),
+///         label: 'Saved',
+///       ),
+///     ],
+///   ),
+/// ),
+/// ```
+class NavigationBar extends StatelessWidget {
+  /// Creates a Material 3 Navigation Bar component.
+  const NavigationBar({
+    Key? key,
+    this.animationDuration = const Duration(milliseconds: 500),
+    this.selectedIndex = 0,
+    required this.destinations,
+    this.onDestinationSelected,
+    this.backgroundColor,
+    this.height,
+    this.labelBehavior,
+  }) :  assert(destinations != null && destinations.length >= 2),
+        assert(0 <= selectedIndex && selectedIndex < destinations.length),
+        super(key: key);
+
+  /// Determines how long the animation is for each destination as it goes from
+  /// unselected to selected.
+  final Duration animationDuration;
+
+  /// Determines which destination from [destinations] is currently selected.
+  ///
+  /// When this is updated, the destination (from [destinations]) at
+  /// [selectedIndex] goes from unselected to selected. It will animate its
+  /// [NavigationDestinationInfo.selectedAnimation] from 0.0 to 1.0.
+  final int selectedIndex;
+
+  /// The list of destinations (usually [NavigationDestination]s) in this
+  /// [NavigationBar].
+  ///
+  /// When [selectedIndex] is updated, the destination from this list at
+  /// [selectedIndex] will animate from 0 (unselected) to 1.0 (selected). When
+  /// the animation is increasing or completed, the destination is considered
+  /// selected, when the animation is decreasing or dismissed, the destination
+  /// is considered unselected.
+  ///
+  /// Any widget in this list will have access to a
+  /// [NavigationDestinationInfo] inherited widget.
+  final List<Widget> destinations;
+
+  /// Called when one of the [destinations] is selected.
+  ///
+  /// This callback usually updates the int passed to [selectedIndex].
+  final ValueChanged<int>? onDestinationSelected;
+
+  /// The color of the [NavigationBar] itself.
+  ///
+  /// If null, [NavigationBarThemeData.backgroundColor] is used. If that
+  /// is also null, the default blends [ColorScheme.surface] and
+  /// [ColorScheme.onSurface] using an elevation overlay.
+  final Color? backgroundColor;
+
+  /// The height of the [NavigationBar]'s Material, from the bottom of the
+  /// screen to the top border of the widget, not including safe area padding
+  /// for things like system navigation.
+  ///
+  /// If null, [NavigationBarThemeData.height] is used. If that
+  /// is also null, the default is 80.
+  final double? height;
```

Comment:
The spec will specify that the nav bar is not affected by visualDensity or textScaleFactor (the text scale factor does however affect the tooltips displayed on long press). So a fixed height is fine

Reply:
This would be a good place to add an explanation of the bar's height:

- It does not adapt to its content
- It does not respond to text scale factor or visual density
- Its value is fixed and the default is ...
- It's fixed because..